docs/languages/csharp-properties.md

```diff
@@ -0,0 +1,253 @@
+
+```
+public class Person
+{
+	public string FirstName
+	{
+		get { return firstName; }
+		set { firstName = value; }
+	}
+	private string firstName;
+	// elided
+}
+```
+ 
+The property definition shown above is a read-write property. Notice
+the keyword `value` in the set accessor. The `set` accessor always has
+a single parameter named `value`. The `get` accessor must return a value
+that is convertable to the type of the property (`string` in this example).
+ 
+That's the basics of the syntax. There are many different variations that support
+a variety of different design idioms. Let's explore those, and learn the syntax
+options for each. 
+
+# Scenarios
+
+The examples above showed one of the simplest cases of property definition:
+a read-write property with no validation. By writing the code you want in the
+`get` and `set` accessors, you can create many different scenarios.  
+
+## Validation
+
+You can write code in the `set` accessor to ensure that the values represented
+by a property are always valid. For example, suppose one rule for the `Person`
+class is that the name cannot be blank, or whitespace. You would write that as
+follows:
+
+```
+public class Person
+{
+    public string FirstName
+    {
+        get { return firstName; }
+        set
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("First name must not be blank");
+            firstName = value;
+        }
+    }
+    private string firstName;
+    // elided
+}
+```
+
+The example above enforces the rule that the first name must not be blank,
+or whitespace. If a developer writes
+```
+foo.FirstName = "";
+```
+That assignment throws an `ArgumentException`. Because a property set accessor
+must have a void return type, you report errors in the set accessor by throwing an exception.
+
+That is a simple case of validation. You can extend this same syntax to anything needed
+in your scenario: You can check the relationships between different properties, or validate
+against any external conditions. Any valid C# statements are valid in a property accessor.
+
+## INotifyPropertyChanged
+
+Another common scenario where you need to write code in a property accessor is to
+support the `INotifyPropertyChanged` interface used in XAML based data binding.
+When the value of a property changes, the object raises the `PropertyChanged` event
+to indicate the change. The data binding libraries, in turn, update display elements
+based on that change. The code below shows how you would implement `INotifyPropertyChanged`
+for the `FirstName` property of this person class.
+
+```
+public class Person : INotifyPropertyChanged
+{
+    public string FirstName
+    {
+        get { return firstName; }
+        set
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("First name must not be blank");
+                if (value != firstName)
```

Comment:
Formatting here is off, making this `if()` statement appear as if it were a child of the above `if()` statement.  I'd also considering using braces just because the one-line conditional statement is long enough to warrant a line break.